store/redis_supplier_roles.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package store
+
+import (
+	"context"
+	"fmt"
+
+	l4g "github.com/alecthomas/log4go"
+
+	"github.com/mattermost/mattermost-server/model"
+)
+
+func (s *RedisSupplier) RoleSave(ctx context.Context, role *model.Role, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
+	if err := s.client.Del(buildRedisKeyForRoleName(role.Name)).Err(); err != nil {
+		l4g.Error("Redis failed to remove key roles:" + role.Name + " Error: " + err.Error())
+	}
+	return s.Next().RoleSave(ctx, role, hints...)
+}
+
+func (s *RedisSupplier) RoleGet(ctx context.Context, roleId string, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
+	// Roles are cached by name, as that is most commonly how they are looked up.
+	// This means that no caching is supported on roles being looked up by ID.
+	return s.Next().RoleGet(ctx, roleId, hints...)
+}
+
+func (s *RedisSupplier) RoleGetByName(ctx context.Context, name string, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
```

Comment:
I'm not sure here, but It can be done by RoleGetByNames(context, [name], hints)? or move part of the logic to another function, like "getFromRedisOrNil(name)" and replace the duplication in both functions?

Reply:
I don't thin it's worth it, because it's only the printing of the error that we save.